source/auth/auth.rst

```diff
@@ -155,6 +157,9 @@ handshake:
     #. A single invalid credential is the same as all credentials being
            invalid.
 
+If the authentication handshake fails for a socket, drivers MUST close all
+other general-use sockets connected to the same server.
```

Comment:
Again, an ok: 0 response to "isMaster"?

Reply:
Yes.  Trying to do mechanism negotiation for a non-existent user will result in an `ok:0` isMaster response.